### DISCUSSION TITLE
Image interface and display for video files (rc.75)

### DISCUSSION THREAD

**Discussion by domdus** [directus/directus#6280] [Archive]
Hi, 

I have collection where items contain a file which is a video (mp4). Before rc.75 I was using the interface "image" even if it's not intended for videos. (name says it)

With the magnifier (Zoom) icon it actually did open a video player to preview/review the video src right there! (on item detail)
Me and the editors liked that very much, as otherwise you can only lookup the referenced video in the file library and watch it there (longer workflow).

Starting with rc.75/rc.76 the image interface display is blank if the file is a video.
![image](https://user-images.githubusercontent.com/67951828/122051210-9f131e80-cde4-11eb-9a6a-9dddeab3e6ee.png)
Any chance to get this "kind of video support" back in? (not naming an interface for video B-) )

**Additional note (and suggestion maybe):**
I would be fine with using the standard file interface which has the "Edit" icon to open the file details, but here no video player is available at all.

1. Edit File
![image](https://user-images.githubusercontent.com/67951828/122052114-9f5fe980-cde5-11eb-807b-7e673b12e449.png)

2. File Interface - Edit view (without player):
![image](https://user-images.githubusercontent.com/67951828/122052246-c4ecf300-cde5-11eb-9871-0c2d8f38f146.png)

**Comment by benhaynes** [comment]
Interesting! Videos should _certainly_ be considered first-class citizens in terms of these asset interfaces/displays. In fact, I would love to give videos their own animated thumbnails on hover. Perhaps we should consider renaming `image` to `media`? The only issue is that I don't think media has a good singular vs plural... but I think we're using the normal M2M for multiple now anyways. @rijkvanzanten? 

And yes, any drawer for DIrectus Files should also include the full-width media preview at the top.

**Comment by philleepflorence** [comment]
How about streamlined file interfaces: 

- `file-image` interface for - [`<picture>`: The Picture Element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/picture)
- `file-video` interface for - [`<video>`: The Video Embed Element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video)
- `file-audio` interface for - [`<audio>`: The Audio Embed Element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/audio)
- `file` interface for - [`<object>`: The Object Embed Element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/object)

In my pipeline for D9 Extensions.

**Comment by loongmxbt** [comment]
Hi, I noticed there is a video player in file management, but not in the file-upload interface. I managed to made a `file video upload` interface based on `file image upload` to provide video preview, but it requires a lot internal functions. 
So what is current situation of `video upload interface`. Hope to have a built-in video interface with preview functionality.
![image](https://user-images.githubusercontent.com/5986600/166624215-0f51a81f-3e7a-488a-af39-b23b330171fe.png)

**Comment by phazonoverload** [comment]
I know it's 4 days shy of 3 years on from this, but just a note that we recently released a new video player interface via the Directus Marketplace. It supports Vimeo/YouTube videos as well as selecting a file from Directus Files. You can then play it directly within Directus Editor 😄 

![image](https://github.com/directus/directus/assets/1461554/4b1eb869-133d-45fe-a681-2dab7124a8be)
